### PR TITLE
Revise documentation for Docker LABEL.

### PR DESCRIPTION
**Pull request by charleswhchan**
Signed-off-by: Charles Chan charleswhchan@users.noreply.github.com

### CODE REVIEW

File: docs/userguide/labels-custom-metadata.md [hunk 1]
```diff
@@ -11,34 +11,34 @@ parent = "mn_use_docker"
 # Apply custom metadata
 
 You can apply metadata to your images, containers, or daemons via
-labels. Metadata can serve a wide range of uses. Use labels to add notes or
-licensing information to an image or to identify a host.
+labels. Labels can serve a wide range of uses cases, it can be use to specify
```
Comment (SvenDowideit) [inline docs/userguide/labels-custom-metadata.md]: s/it/they/ :) plural

File: docs/userguide/labels-custom-metadata.md [hunk 2]
```diff
@@ -11,34 +11,34 @@ parent = "mn_use_docker"
...
+notes or licensing information to an image or to identify a host.
 
 A label is a `<key>` / `<value>` pair. Docker stores the label values as
-*strings*. You can specify multiple labels but each `<key>` / `<value>` must be
-unique to avoid overwriting. If you specify the same `key` several times but with
-different values, newer labels overwrite previous labels. Docker uses
-the last `key=value` you supply.
+*strings*. You can specify multiple labels but `<key>` must be
```
Comment (SvenDowideit) [inline docs/userguide/labels-custom-metadata.md]: either `each <key>` or `the <key>`

File: docs/userguide/labels-custom-metadata.md [hunk 3]
```diff
@@ -147,7 +146,7 @@ You can view the labels via the `docker inspect` command:
 ## Query labels
 
 Besides storing metadata, you can filter images and containers by label. To list all
-running containers that the `com.example.is-beta` label:
+running containers that contains the `com.example.is-beta` label:
```
Comment (SvenDowideit) [inline docs/userguide/labels-custom-metadata.md]: mmm, probably better to replace `container` with `have`

File: docs/userguide/labels-custom-metadata.md [hunk 4]
```diff
@@ -156,7 +155,7 @@ List all running containers with a `color` label of `blue`:
 
     $ docker ps --filter "label=color=blue"
 
-List all images with `vendor` `ACME`:
+List all images with `vendor` label of `ACME`:
```
Comment (SvenDowideit) [inline docs/userguide/labels-custom-metadata.md]: List all images with the label `vendor` that have the value `ACME` ?

**Comment by SvenDowideit** [comment]
nice job thank you :) I've a few little nits.

File: docs/userguide/labels-custom-metadata.md (hunk 1 above)
Comment (charleswhchan) [inline docs/userguide/labels-custom-metadata.md]: Fixed.

File: docs/userguide/labels-custom-metadata.md (hunk 2 above)
Comment (charleswhchan) [inline docs/userguide/labels-custom-metadata.md]: Fixed.

File: docs/userguide/labels-custom-metadata.md (hunk 3 above)
Comment (charleswhchan) [inline docs/userguide/labels-custom-metadata.md]: Yeah, that is less confusion. :smile:

File: docs/userguide/labels-custom-metadata.md (hunk 4 above)
Comment (charleswhchan) [inline docs/userguide/labels-custom-metadata.md]: Fixed. (Also update the description a few lines above similarly)

**Comment by charleswhchan** [comment]
Thanks for the review. Nits fixed. :smiley:

**Comment by charleswhchan** [comment]
ping: Anything additional comments?

File: docs/userguide/labels-custom-metadata.md [hunk 5]
```diff
@@ -11,34 +11,34 @@ parent = "mn_use_docker"
 # Apply custom metadata
 
 You can apply metadata to your images, containers, or daemons via
-labels. Metadata can serve a wide range of uses. Use labels to add notes or
-licensing information to an image or to identify a host.
+labels. Labels can serve a wide range of uses cases, they can be use to specify
```
Comment (thaJeztah) [inline docs/userguide/labels-custom-metadata.md]: s/uses cases/uses/  (it's not a use *case* here)
s/can be use/can be used/

Alternatively;

```
Labels serve a wide range of uses, such as adding notes or licensing information
to an image, or to identify a host.
```

File: docs/userguide/labels-custom-metadata.md [hunk 6]
```diff
@@ -11,34 +11,34 @@ parent = "mn_use_docker"
...
 >**Note:** Support for daemon-labels was added in Docker 1.4.1. Labels on
 >containers and images are new in Docker 1.6.0
 
 ## Label keys (namespaces)
 
-Docker puts no hard restrictions on the label `key` you. However, labels with
-simple keys can conflict. For example, you can categorize your images by using a
-chip "architecture" label:
+Docker puts no hard restrictions on the `key` used for a label. However, using
+simple keys can easily leads to conflicts. As an example, let's say you
```
Comment (thaJeztah) [inline docs/userguide/labels-custom-metadata.md]: s/easily leads/easily lead/

Comment (thaJeztah) [inline docs/userguide/labels-custom-metadata.md]: I think "As an example, let's say you" is saying the same thing twice, maybe just;

```
For example, you have chosen to categorize your images by CPU architecture
using "architecture" labels in your Dockerfiles:

File: docs/userguide/labels-custom-metadata.md [hunk 7]
```diff
@@ -87,31 +86,31 @@ You can store this struct in a label by serializing it to a string first:
...
+To add labels to an image, use the `LABEL` instruction in your Dockerfile:
 
 
     LABEL [<namespace>.]<key>[=<value>] ...
 
-The `LABEL` instruction adds a label to your image, optionally setting its value.
+The `LABEL` instruction adds a label to your image, optionally with a value.
 Use surrounding quotes or backslashes for labels that contain
-white space character:
+white space character in the `<value>`:
```
Comment (thaJeztah) [inline docs/userguide/labels-custom-metadata.md]: character should be plural here (characters)

File: docs/userguide/labels-custom-metadata.md [hunk 8]
```diff
@@ -59,15 +59,14 @@ notation. Use the following guidelines to name your keys:
   cumbersome namespaces on the command-line.
 
 
-These are guidelines and Docker does not *enforce* them. Failing following these
-guidelines can result in conflicting labels. If you're building a tool that uses
-labels, you *should* use namespaces for your label keys.
+These are simply guidelines and Docker does not *enforce* them. However, for the
+the benefit of the community, you *should* use namespaces for your label keys.
```
Comment (thaJeztah) [inline docs/userguide/labels-custom-metadata.md]: There's one "the" too many here (one at the start of this line, one at the end of the previous line)

File: docs/userguide/labels-custom-metadata.md (hunk 6 above)
Comment (charleswhchan) [inline docs/userguide/labels-custom-metadata.md]: :sweat_smile:

**Comment by charleswhchan** [comment]
@thaJeztah Thanks the picking out the mistakes. PR updated.

**Comment by thaJeztah** [comment]
No worries, they were just some 'nits'.

LGTM

ping @SvenDowideit PTAL

**Comment by bboreham** [comment]
I noticed this page announces that labels can be set on containers, but adds nothing further on how you might achieve this. Should I create a separate issue (PR, maybe), or might you incorporate it in this one?

**Comment by bboreham** [comment]
Oops, never mind - https://github.com/docker/docker/pull/16449

**Comment by thaJeztah** [comment]
ping @fredlf @jamtur01 @moxiegirl @SvenDowideit PTAL, thanks!

**Comment by jamtur01** [comment]
I would wrap Dockerfile in code quotes but otherwise looks good to me.

File: docs/userguide/labels-custom-metadata.md [hunk 9]
```diff
@@ -59,15 +60,14 @@ notation. Use the following guidelines to name your keys:
...
-labels, you *should* use namespaces for your label keys.
+These are simply guidelines and Docker does not *enforce* them. However, for
+the benefit of the community, you *should* use namespaces for your label keys.
 
 
 ## Store structured data in labels
 
-Label values can contain any data type that can be stored as a string. For
-example, consider this JSON:
+Label values can contain any data type as long as it can be represented as a
```
Comment (moxiegirl) [inline docs/userguide/labels-custom-metadata.md]: @charleswhchan This isn't adding clarity the antecedent `it` is unclear.  Why did you want to change this?

File: docs/userguide/labels-custom-metadata.md [hunk 10]
```diff
@@ -87,31 +87,31 @@ You can store this struct in a label by serializing it to a string first:
 While it is *possible* to store structured data in label values, Docker treats
 this data as a 'regular' string. This means that Docker doesn't offer ways to
 query (filter) based on nested properties. If your tool needs to filter on
-nested properties, the tool itself should implement this.
+nested properties, the tool itself needs to implement this functionality.
```
Comment (moxiegirl) [inline docs/userguide/labels-custom-metadata.md]: Same comment here.

**Comment by charleswhchan** [comment]
@moxiegirl It's been too long and the reason escaped my mind right now... :smile: 
No big deal, reverted the 2 incidents and updated the PR.

**Comment by moxiegirl** [comment]
Thanks @charleswhchan LGTM! And go!

**Comment by charleswhchan** [comment]
Thanks.

**Pull request merged**